external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -118,17 +122,33 @@ fn extract_macros(context: &mut Context, modules: &UniqueMap<ModuleIdent, N::Mod
         }
         macro_use_funs
     }
-    let all_macro_definitions = modules.ref_map(|_mident, mdef| {
-        mdef.functions.ref_filter_map(|_name, f| {
-            let _macro_loc = f.macro_?;
-            if let N::FunctionBody_::Defined((use_funs, body)) = &f.body.value {
-                let use_funs = merge_use_funs(&mdef.use_funs, use_funs.clone());
-                Some((use_funs, body.clone()))
-            } else {
-                None
-            }
+
+    fn extract_modules_macros(
+        macros_modules: &UniqueMap<ModuleIdent, N::ModuleDefinition>,
+    ) -> UniqueMap<ModuleIdent, UniqueMap<FunctionName, N::Sequence>> {
+        macros_modules.ref_map(|_mident, mdef| {
+            mdef.functions.ref_filter_map(|_name, f| {
+                let _macro_loc = f.macro_?;
+                if let N::FunctionBody_::Defined((use_funs, body)) = &f.body.value {
+                    let use_funs = merge_use_funs(&mdef.use_funs, use_funs.clone());
+                    Some((use_funs, body.clone()))
+                } else {
+                    None
+                }
+            })
         })
-    });
+    }
+
+    let mut all_macro_definitions = extract_modules_macros(modules);
+
+    // Prefer local bindings to previous ones. This is ostensibly an error, but naming should have
+    // already produced that error. To avoid unnecessary error handling, we simply prefer the
+    // non-precompiled definitions.
+    if let Some(libs) = pre_compiled_lib {
+        let lib_macros = extract_modules_macros(&libs.naming.inner.modules);
+        all_macro_definitions =
+            all_macro_definitions.union_with(&lib_macros, |_key, local, _previous| local.clone());
```

Comment:
I feel like we should be able to remove this clone? not a huge deal though but might hurt perf it actually ends up cloning